modules/internal/ChapelBase.chpl

```diff
@@ -1032,20 +1032,23 @@ module ChapelBase {
     return if x != 0i then true else false;
 
   pragma "dont disable remote value forwarding"
+  pragma "no copy return"
```

Comment:
It needs a "no copy return" because it may be used to initialize an array field (from what I could tell while debugging leaks/bugs).

You're probably right about the param and nilType versions, I likely just applied it to all of them without thinking much. I'll remove them and add a comment.